btcutil/amount.go

```diff
@@ -101,11 +102,23 @@ func (a Amount) ToBTC() float64 {
 
 // Format formats a monetary amount counted in bitcoin base units as a
 // string for a given unit.  The conversion will succeed for any unit,
-// however, known units will be formated with an appended label describing
+// however, known units will be formatted with an appended label describing
 // the units with SI notation, or "Satoshi" for the base unit.
 func (a Amount) Format(u AmountUnit) string {
 	units := " " + u.String()
-	return strconv.FormatFloat(a.ToUnit(u), 'f', -int(u+8), 64) + units
+	formatted := strconv.FormatFloat(a.ToUnit(u), 'f', -int(u+8), 64)
+
+	// when formatting full BTC, add trailing zeroes for numbers with decimal
+	// point to ease reading of sat amount
+	if u == AmountBTC {
+		decimalPoint := "."
+		before, after, containsDecimalPoint := strings.Cut(formatted, decimalPoint)
```

Comment:
`strings.Cut` was only added in Go `1.18`, but our `go.mod` still says `Go 1.16`.
Also, why not just:
```
		if strings.Contains(formatted, ".") {
			return fmt.Sprintf("%.8f%s", a.ToUnit(u), units)
		}
```